solution-workshop/src/main/java/io/axoniq/demo/hotel/booking/query/RoomAvailabilityEntityResolver.java

```diff
@@ -0,0 +1,45 @@
+package io.axoniq.demo.hotel.booking.query;
+
+import io.axoniq.demo.hotel.booking.command.api.RoomEvent;
+import org.axonframework.messaging.Message;
+import org.axonframework.messaging.annotation.ParameterResolver;
+import org.axonframework.messaging.annotation.ParameterResolverFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.lang.reflect.Executable;
+import java.lang.reflect.Parameter;
+
+@Component
+public class RoomAvailabilityEntityResolver implements ParameterResolver<RoomAvailabilityEntity>,
+        ParameterResolverFactory {
+
+    private final RoomAvailabilityEntityRepository roomAvailabilityEntityRepository;
+
+    @Autowired
+    public RoomAvailabilityEntityResolver(RoomAvailabilityEntityRepository roomAvailabilityEntityRepository) {
+        this.roomAvailabilityEntityRepository = roomAvailabilityEntityRepository;
+    }
+
+    @Override
+    public RoomAvailabilityEntity resolveParameterValue(Message<?> message) {
+        if (matches(message)) {
```

Comment:
This `matches` call is not necessary, right? When this is invoked the matches call has already been invoked by the caller